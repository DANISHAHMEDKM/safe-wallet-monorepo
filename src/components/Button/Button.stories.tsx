import React from 'react'
import { View } from 'react-native'
import type { Meta, StoryObj } from '@storybook/react'
import { SafeButton } from './Button'

const meta = {
  title: 'SafeButton',
  component: SafeButton,
  argTypes: {
    onPress: { action: 'pressed the button' },
  },
  args: {
    text: 'Hello world',
  },
  decorators: [
    (Story) => (
      <View style={{ padding: 16, alignItems: 'flex-start' }}>
        <Story />
      </View>
    ),
  ],
} satisfies Meta<typeof SafeButton>

export default meta

type Story = StoryObj<typeof meta>

export const Basic: Story = {}

export const AnotherExample: Story = {
  args: {
    text: 'Another example',
  },
}
