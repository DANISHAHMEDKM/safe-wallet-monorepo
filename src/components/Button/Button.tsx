import { TouchableOpacity, Text, StyleSheet } from "react-native";

export interface SafeButtonProps {
  onPress?: () => void;
  text: string;
}

export const SafeButton = ({ onPress, text }: SafeButtonProps) => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={onPress}
      activeOpacity={0.8}
    >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: "purple",
    borderRadius: 8,
  },
  text: { color: "white" },
});
