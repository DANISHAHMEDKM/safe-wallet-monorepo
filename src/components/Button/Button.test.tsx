import { vi } from 'vitest'
import { render, userEvent } from '@/src/tests/test-utils'
import { SafeButton } from './Button'

describe('Button', () => {
  beforeEach(() => {
    vi.clearAllMocks()
  })

  it('should render the default markup', () => {
    const container = render(<SafeButton onPress={vi.fn()} text="sign" />)

    expect(container).toMatchSnapshot()
  })

  it('should be able to click', async () => {
    const user = userEvent.setup()

    const mockedFn = vi.fn()
    const container = render(<SafeButton onPress={mockedFn} text="sign" />)

    await user.press(container.getByText('sign'))

    expect(mockedFn).toHaveBeenCalled()
  })
})
