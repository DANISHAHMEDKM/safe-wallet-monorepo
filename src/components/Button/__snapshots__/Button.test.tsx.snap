// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the default markup 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "purple",
      "borderRadius": 8,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
      }
    }
  >
    sign
  </Text>
</View>
`;
